src/Graphics/Model.elm

```diff
@@ -0,0 +1,97 @@
+module Graphics.Model where
+
+import Color (Color, Gradient)
+import Maybe (Maybe)
+import Transform2D (Transform2D)
+
+type alias Form =
+    { theta : Float
+    , scale : Float
+    , x : Float
+    , y : Float
+    , alpha : Float
+    , form : BasicForm
+    }
+
+type BasicForm
+    = FPath LineStyle Path
+    | FShape ShapeStyle Shape
+    | FImage Int Int (Int,Int) String
+    | FElement Element
+    | FGroup Transform2D (List Form)
+
+type alias Path = List (Float,Float)
+
+type alias Shape = List (Float,Float)
+
+{-| The shape of the ends of a line. -}
+type LineCap = Flat | Round | Padded
```

Comment:
So far, this PR cannot address the latter though because the public documentation will not work out. Why should we do wrong things as stopgaps, especially if they do not really work? I am okay with some extra code size for a little while compared to the changes we are considering here.